### PR TITLE
Hub Menu: Disable multiple taps on menu items

### LINKED ISSUE #6436: Handle multiple taps in Hub Menu
There is one issue with the Menu tab – it looks like you can tap multiple buttons at the same time, firing off multiple modal windows. It looks like `exclusiveTouch` needs to be set on the container View to prevent that from happening, but on SwiftUI this property is not existing at all, so we should try some workaround to handle that.

https://user-images.githubusercontent.com/495617/158427483-9535384d-f044-452b-ac5e-3bb51024aff7.MOV

### PR DESCRIPTION
**Pull request by itsmeichigo**
Closes: #6436 

### Description

This PR adds a workaround to disable multiple taps on the items of the Hub Menu.

The solution is:
- Keep a state in the hub menu for whether the menu items should be disabled temporarily
- Send the state as a binding property to each item - the items will have to update this value to false when its button receives a tap event.
- Reset the state to make sure all items are tappable after the view appears again / when modals are dismissed.

The downside of this solution is that there's a potential case when the tap state is not reset properly causing the menu items to be disabled. `onAppear` isn't called when modals are dismissed, so we need to take care of resetting the selection state on the dismissal of every modal. 

### Testing instructions

- Launch the app and navigate to the Menu tab.
- Try tapping on any item more than one time or tapping on more items at a time.
- Notice that only one new flow is presented at a time.

### Screenshots

https://user-images.githubusercontent.com/5533851/159624770-bcbcb8b3-f02f-499f-a155-9cb1c70f851b.mov

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by pmusolino** [comment]
> The downside of this solution is that there's a potential case when the tap state is not reset properly causing the menu items to be disabled. onAppear isn't called when modals are dismissed, so we need to take care of resetting the selection state on the dismissal of every modal.

This looks a potential downside, which is worse than the current bug. Do you think it's worth merging this PR given this other potential issue?

**Review by pmusolino — COMMENTED** [review]
It seems that the bug is still present, tapping the two elements on top

https://user-images.githubusercontent.com/495617/159694332-fc18e447-2083-4e5d-a329-215ed7786a4a.mp4

**Comment by itsmeichigo** [comment]
Hi, I've added a new change to prevent the execution of the tap gesture when the menu item is disabled [here](cd39a7fe395d337e0b8a541d27ab7740973a2701). This is necessary to prevent multiple taps on different items - the `disabled` modifier may not be sufficient because it requires the view to be reloaded to take effect.

> This looks a potential downside, which is worse than the current bug. Do you think it's worth merging this PR given this other potential issue?

In my opinion, the issue is highly noticeable during the development phase and therefore less likely to slip into production. But if we can come up with a scenario that proves otherwise, I'm happy to close this PR.

**Comment by pmusolino** [comment]
> In my opinion, the issue is highly noticeable during the development phase and therefore less likely to slip into production. But if we can come up with a scenario that proves otherwise, I'm happy to close this PR.

I leave the floor to @astralbodies  who was the first to notice the problem. What do you think?

**Comment by astralbodies** [comment]
> I leave the floor to @astralbodies who was the first to notice the problem. What do you think?

I'm not sure what to think here - I guess you could always set a one-second timer to unset that state. 

I was also looking at "exclusively" in the SwiftUI documentation. Is that relevant here at all?

https://developer.apple.com/documentation/swiftui/gesture/exclusively(before:)

**Comment by itsmeichigo** [comment]
I like the idea of using timer to reset the disabled state - updated it in 91ddb36ab0dae42ae2a9a15f5f410644728f7c8c.

> I was also looking at "exclusively" in the SwiftUI documentation. Is that relevant here at all?

If I understand it correctly, it seems that `exclusively` is more suitable for combining different gestures on the same view. In our case, we have to handle the selection state of different views (menu items) simultaneously so I don't think that method can be applied here.

File: WooCommerce/Classes/ViewRelated/Hub Menu/HubMenu.swift [hunk 1]
```diff
@@ -13,6 +13,12 @@ struct HubMenu: View {
     @State private var showingReviews = false
     @State private var showingCoupons = false
 
+    /// State to disable multiple taps on menu items
+    /// Make sure to reset the value to false after dismissing sub-flows
+    @State private var shouldDisableItemTaps = false
+
+    private let timer = Timer.publish(every: 1, on: .main, in: .common).autoconnect()
```
Comment (pmusolino) [inline WooCommerce/Classes/ViewRelated/Hub Menu/HubMenu.swift:22]: What do you think about adding a comment about this timer?

**Review by pmusolino — APPROVED** [review]
> I like the idea of using timer to reset the disabled state - updated it in https://github.com/woocommerce/woocommerce-ios/commit/91ddb36ab0dae42ae2a9a15f5f410644728f7c8c.

The solution works great! :shipit:

File: WooCommerce/Classes/ViewRelated/Hub Menu/HubMenu.swift (hunk 1 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Hub Menu/HubMenu.swift:22]: Sure - updated in f0cb2cb0536099928614f1fdd8943cc76a9234db.

**Pull request merged**